File: java/src/main/java/com/sasteval/vuln/ailegacy/HardcodedSecrets.java
package com.sasteval.vuln.ailegacy;

import javax.servlet.ServletException;
import javax.servlet.http.HttpServlet;
import javax.servlet.http.HttpServletRequest;
import javax.servlet.http.HttpServletResponse;
import java.io.IOException;
import java.nio.charset.StandardCharsets;
import java.util.Base64;

/**
 * CWE-798: Use of Hard-coded Credentials
 * VULNERABILITY: JWT secret and API key are hardcoded in source code.
 */
public class HardcodedSecrets extends HttpServlet {

    // VULN: Hardcoded secrets in source code
    private static final String JWT_SECRET = "super-secret-jwt-key-12345";
    private static final String API_KEY = "AKIA_EXAMPLE_KEY_DO_NOT_USE";

    @Override
    protected void doGet(HttpServletRequest request, HttpServletResponse response)
            throws ServletException, IOException {

        String username = request.getParameter("username");
        response.setContentType("application/json");

        // VULN: Using hardcoded secret to "sign" a token
        String header = Base64.getEncoder().encodeToString(
                "{\"alg\":\"HS256\",\"typ\":\"JWT\"}".getBytes(StandardCharsets.UTF_8));
        String payload = Base64.getEncoder().encodeToString(
                ("{\"sub\":\"" + username + "\",\"admin\":true}").getBytes(StandardCharsets.UTF_8));
        String signature = Base64.getEncoder().encodeToString(
                (header + "." + payload + JWT_SECRET).getBytes(StandardCharsets.UTF_8));

        String token = header + "." + payload + "." + signature;

        response.getWriter().println("{\"token\": \"" + token + "\"}");
    }
}

File: dotnet/Vuln/AiLegacy/HardcodedSecrets.cs
namespace SastEval.Vuln.AiLegacy;

/// <summary>
/// CWE-798: Use of Hard-coded Credentials - Secrets embedded directly in source code.
/// </summary>
public static class HardcodedSecrets
{
    // VULNERABLE: Hard-coded JWT signing key in source code
    private const string JwtSecret = "super-secret-jwt-key-12345";

    // VULNERABLE: Hard-coded API key in source code
    private const string ApiKey = "AKIA_EXAMPLE_KEY_DO_NOT_USE";

    public static IResult Handle(HttpContext context)
    {
        var action = context.Request.Query["action"].ToString();

        if (action == "verify")
        {
            return Results.Ok(new
            {
                Message = "Token verified using internal signing key.",
                KeyPrefix = JwtSecret[..5] + "..."
            });
        }

        if (action == "api-call")
        {
            return Results.Ok(new
            {
                Message = "External API called.",
                KeyUsed = ApiKey[..8] + "..."
            });
        }

        return Results.Ok(new { Message = "Specify action=verify or action=api-call" });
    }
}
